### DISCUSSION TITLE
WSA: Context3D not available

### DISCUSSION THREAD

**Discussion by jenatali** [airsdk/Adobe-Runtime-Support#1860] [Q&A] [answered]
Hi,

I’m a developer working on the Windows Subsystem for Android™️ (WSA), and I’m looking at running some Adobe AIR applications that are using Starling, but it looks like they’re all getting “Context3D not available” (error 3702). This seems to be a problem in our graphics stack, as if I swap out the GLES/EGL implementation for a different one (that’s not generally ship-quality), the apps run.

Any chance you’d be able to provide some pointers for things that might be going wrong in this environment? I’m not seeing any errors coming out of GLES/EGL as far as I can tell, and it seems that any capabilities that’re queried are all within reason. This device doesn’t have the 4x MSAA option enabled either.

**Comment by ajwfrost** [comment, marked as answer]
Hi

I've just been trying to see if we have a suitable machine for getting WSA up and running here .. I think our company machines fall down at the requirement for SSD devices! It will take a bit of time to procure these...

So in the meantime -> could we send you a custom runtime that will give more information on what's failing? There are a fair number of steps and conditions involved so I'm not sure what might be the problem here. Presumably you're packaging and running an application as if it's just a normal Android application - would you be able to pick up logcat messages if we put in some __android_log_print() NDK calls?

thanks

**Reply by jenatali** [reply]
That'd be incredibly helpful, yes. So far, I've just been running pre-packaged 3rd party apps where I've been seeing this error, and haven't really tried building my own - I got as far as downloading the AIR SDK, but got a bit overwhelmed with technology that I'm very unfamiliar with. If you were able to provide a dead-simple pre-packaged APK with that logging, that would be fantastic.

The graphics stack should be GLES3.0 capable, so I'd expect it to be able to run with pretty much any AIR Context3D profile, and all the logging I've added myself shows lots of context creation and destruction, so presumably these apps are using 'auto' and not succeeding at even the most baseline profiles.

Feel free to follow up over email if necessary. Thanks!

**Reply by jenatali** [reply]
Andrew and I went back-and-forth offline a bit and settled on it: When an AIR app requests depth and stencil to be disabled, AIR requires a config to be present that has no depth/stencil bits.
